drivers/cc110x/cc110x_netdev.c

```diff
@@ -353,13 +353,23 @@ static int cc110x_init(netdev_t *netdev)
         gpio_irq_disable(dev->params.gdo2);
         DEBUG("[cc110x] netdev_driver_t::init(): cc110x_apply_config() "
               "failed\n");
-        /* Pass through received error code  */
+        /* Pass through received error code */
         return retval;
     }
-    else {
-        DEBUG("[cc110x] netdev_driver_t::init(): Success\n");
+
+    if (CC110X_DEFAULT_CHANNEL) {
```

Comment:
Does it make sense to make this conditional?
I see that the channel is probably 0 after reset, but if we always exercise this code paths errors would be caught early on.
Also init shouldn't be faster just because you use a certain channel ;)

Reply:
All configuration registers are uploaded in a "burst write" (as the data sheet calls it), which is only possible for consecutive registers. In these burst writes only for the first register the address has to be written, which greatly improves configuration speed. One of these configuration registers it the channel register.

It would be super neat to have the default channel in this configuration. But this sadly is not possible, as RIOT's channel number differs from the physical channel number. (There is a simple reason for that: The device has strict constraints on the size of channels. If big channel spacings are used, one has to only use every n-th of the available channels. For that reason, a mapping between logical channel numbers (as used in RIOT) and physical channel numbers (as written to the register) has been implemented.) For that reason it is not possible to compute the physical channel number from RIOT's virtual channel number at compile time.

But putting your finger on that made me realize, that there is indeed a bug. If the virtual channel 0 does not map to the physical channel 0, this will not work. For the channel mapping `cc110x_chanmap_433mhz_50khz_alt` this is the case. So I should indeed configure the channel unconditionally.

Reply:
Oh, no it is fine. `cc110x_apply_config()` always switches to channel zero. (The channel mappings have a different number of valid channels, but they all have to contain at least one channel. For that reason, going to channel 0 is the safest option.)

Reply:
I understand that it's not nececcary to call `cc110x_set_channel()` if the default channel is 0. What I'm saying is that I don't think it's a good idea to have `cc110x_init()` behave different depending on the configured default channel.

I'd assume most users (and testers) will not change the default channel and probably also not call `ifconfig set channel` very often.

So if anything breaks down the road, only the user who decided to use a different channel than 0 will notice. (And probably scratch their head about why the behavior is different depending on the channel)

I think those few cycles saved on init should not be worth the hassle (and they occur anyway for every channel != 0).

Reply:
I changed `cc110x_apply_config()` to also expect the new channel number as argument. That way the code path is taken regardless of channel. Also, the channel is now only set once during initialization regardless of channel.